File: src/lessons/sort/bubble/AlgBubbleSort1Entity.java
package lessons.sort.bubble;

import jlm.universe.sort.SortingEntity;

public class AlgBubbleSort1Entity extends SortingEntity {
	
	public void run() {
		this.bubbleSort();
	}
	
	/* BEGIN TEMPLATE */
	public void bubbleSort()  {
		/* BEGIN SOLUTION */
		boolean swapped;
		do {
			swapped = false;
			for (int i=0; i<getValueCount()-1; i++)
				if (!isSmaller(i,i+1)) {
					swap(i,i+1);
					swapped =true;
				}
		} while (swapped);
		checkme(); /* color everything in blue */
		/* END SOLUTION */
	}
	/* END TEMPLATE */

}

File: resources/exercises/sort/basic/bubble/AlgBubbleSort1Entity.c
//RemoteSort

/* BEGIN TEMPLATE */
void bubbleSort()  {
	/* BEGIN SOLUTION */
	int swapped;
	do {
		swapped = 0;
		int i;
		for (i=0; i<getValueCount()-1; i++)
			if (!isSmaller(i,i+1)) {
				swap(i,i+1);
				swapped =1;
			}
	} while (swapped);
	/* END SOLUTION */
}
/* END TEMPLATE */

void run(){
	bubbleSort();
}
